File: java/singleton/loggercomsingleton/src/Logger.java
public class Logger {
    // Atributo estático que mantém a instância única
    private static Logger instance;

    // Construtor privado para evitar a criação de novas instâncias
    private Logger() {
        System.out.println("Instância única do Logger criada!");
    }

    // Método para obter a instância única
    public static Logger getInstance() {
        if (instance == null) {
            instance = new Logger();
        }
        return instance;
    }

    public void log(String message) {
        System.out.println("Log: " + message);
    }
}

File: python/singleton/loggercomsingleton/Logger.py
class Logger():
    # Atributo de classe que armazenará a instância única
    _instance = None

    # Método especial __new__ para controlar a criação da instância
    def __new__(cls):
        if cls._instance is None:
            # Cria a instância caso ainda não exista
            cls._instance = super(Logger, cls).__new__(cls)
            print("Inst\u00e2ncia \u00fanica do Logger criada!")
        return cls._instance
    
    

    # Exemplo de método que pode ser acessado pela instância Singleton
    def log(self, mesage):
        print(f"Log: {mesage}")